bindings/python/pymongoarrow/lib.pyx

```diff
@@ -249,9 +249,9 @@ def process_bson_stream(bson_stream, context, arr_value_builder=None):
                         bson_iter_decimal128(&doc_iter, &dec128)
                         if sys.byteorder == 'little':
                             val = pystruct.pack('<QQ', dec128.low, dec128.high)
+                            builder.append(val)
                         else:
-                            val = pystruct.pack('<QQ', dec128.high, dec128.low)
-                        builder.append(val)
+                            builder.append_null()
```

Comment:
Can we add a comment that says big endian systems are unsupported/untested?

Reply:
I added a note to the README in addition to the specific note about `Decimal128` in `supported_types.rst`.

Reply:
I mean a comment in the code to explain why we do append_null() here.

Reply:
Done